### PR TITLE
add doi argument

### PR DESCRIPTION
**Pull request by florianl**
Signed-off-by: Florian Lehner <dev@der-flo.net>

Add the option to provide a DOI instead of a record ID.

Example:
```
./main list -d 10.7483/OPENDATA.CMS.A342.9982
```

### CODE REVIEW

**Comment by florianl** [comment]
hey @clelange :wave: 
can I get feedback on this PR?

**Comment by clelange** [comment]
Hi @florianl - sorry about the delay. I've had a brief look, and in principle this looks good to me and is a welcomed addition. However, my code (and yours) doesn't have any tests, so the only way for me to check that it's working is to check out your PR and test it manually... I'll get to it some time later this week, maybe even tomorrow.

I'd of course appreciate very much if you implemented some tests, e.g. similar to what's done at:
- https://github.com/cernopendata/cernopendata-client/blob/master/tests/test_cli_download_files.py#L35
- https://github.com/cernopendata/cernopendata-client/blob/master/tests/test_cli_get_file_locations.py#L58
- https://github.com/cernopendata/cernopendata-client/blob/master/tests/test_cli_get_metadata.py#L38

These tests could also come in a separate PR, anything is appreciated 😃

File: cmd/root.go [hunk 1]
```diff
@@ -65,20 +68,104 @@ func validateProtocolChoice() error {
...
+}
+
+func getRecordIDFromDoi() error {
+	params := url.Values{}
+	params.Add("page", "1")
+	params.Add("size", "1")
+	params.Add("q", fmt.Sprintf(`doi:"%s"`, doi))
+
+	queryURL := fmt.Sprintf("%s/api/records?%s", server, params.Encode())
+	fmt.Println(queryURL)
```
Comment (clelange) [inline cmd/root.go]: Please remove debug statement.

**Comment by clelange** [comment]
Your example fails for me:

```
./cernopendata-client-go list -d 10.7483/OPENDATA.CMS.A342.9982
http://opendata.cern.ch/api/records?page=1&q=doi%3A%2210.7483%2FOPENDATA.CMS.A342.9982%22&size=1
Error: failed to get recordID from doi: recordID needs to be positive integer number
```

Can you please check what's wrong?

**Review by clelange — CHANGES_REQUESTED** [review]

File: cmd/root.go (hunk 1 above)
Comment (florianl) [reply cmd/root.go]: my fault :see_no_evil: removed it

**Comment by florianl** [comment]
> Can you please check what's wrong?

I have checked the value `recordID` before assigning it with `recordID = r.Hits.Hits[0].Metadata.Recid`.
The check on the value I have added just before opening the PR :see_no_evil:  this is fixed now

**Review by clelange — APPROVED** [review]
Thanks, now it's working and looks good to me!

**Pull request merged**